pkg/utils/release/controller.go

```diff
@@ -109,29 +114,146 @@ func (s *SuiteController) GetPipelineRunInNamespace(namespace, releaseName, rele
 
-// Get ReleaseLink object from a given namespace
-func (s *SuiteController) GetReleaseLink(name string, namespace string) (*v1alpha1.ReleaseLink, error) {
-	namespacedName := types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
+// CreateReleasePlanAdmission creates a new ReleasePlanAdmission using the given parameters.
+func (s *SuiteController) CreateReleasePlanAdmission(name, source_namespace, application, target_namespace, AutoReleaseLabel string, releaseStrategy string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	var releasePlanAdmission *appstudiov1alpha1.ReleasePlanAdmission
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlanAdmission is not missing
+		releasePlanAdmission = &appstudiov1alpha1.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: target_namespace,
+				Labels: map[string]string{
+					appstudiov1alpha1.AutoReleaseLabel: AutoReleaseLabel,
+				},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanAdmissionSpec{
+				DisplayName: name,
+				Application: application,
+				Origin: kcp.NamespaceReference{
+					Namespace: source_namespace,
+				},
+				Environment:     "test-environment",
+				ReleaseStrategy: releaseStrategy,
+			},
+		}
+	} else { // AutoReleaseLabel in ReleasePlanAdmission is missing
+		releasePlanAdmission = &appstudiov1alpha1.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: target_namespace,
+				Labels:    map[string]string{},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanAdmissionSpec{
+				DisplayName: name,
+				Application: application,
+				Origin: kcp.NamespaceReference{
+					Namespace: source_namespace,
+				},
+				Environment:     "test-environment",
+				ReleaseStrategy: releaseStrategy,
+			},
+		}
 	}
+	return releasePlanAdmission, s.KubeRest().Create(context.TODO(), releasePlanAdmission)
+}
+
+// GetReleasePlan returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlanAdmission(releasePlanAdmissionName, devNamespace string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	releasePlanAdmission := &appstudiov1alpha1.ReleasePlanAdmission{}
+
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      releasePlanAdmissionName,
+		Namespace: devNamespace,
+	}, releasePlanAdmission)
 
-	releaseLink := &v1alpha1.ReleaseLink{}
-	err := s.KubeRest().Get(context.TODO(), namespacedName, releaseLink)
-	if err != nil {
-		return nil, err
+	return releasePlanAdmission, err
+}
+
+// DeleteReleasePlan deletes the releasePlan resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlanAdmission(releasePlanAdmissionName, devNamespace string, reportErrorOnNotFound bool) error {
```

Comment:
can we just call this namespace? The `ReleasePlanAdmission` is not even in the dev namespace anyways... it is in the managed one

Reply:
Then I need to call it (target_namespace) as we use generically the source and target namespaces